api_views/users.py

```diff
@@ -206,3 +206,49 @@ def delete_user(username):
                 return Response(error_message_helper("User not found!"), 404, mimetype="application/json")
         else:
             return Response(error_message_helper("Only Admins may delete users!"), 401, mimetype="application/json")
+
+
+def get_user_by_email(email):
```

Comment:
The get_user_by_email function uses a generic error message when user is not found, making it unclear whether the error is due to invalid email format or non-existent user